File: event-service/src/main/java/com/greenlight/event/service/EventStatusService.java
package com.greenlight.event.service;

import com.greenlight.event.domain.EventStatus;
import com.greenlight.event.repository.EventStatusRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@Slf4j
@Service
@RequiredArgsConstructor
public class EventStatusService implements ReadOnlyService<EventStatus, Integer> {
  private final EventStatusRepository eventStatusRepository;

  @Override
  public Flux<EventStatus> findAll() {
    return eventStatusRepository.findAll();
  }

  @Override
  public Mono<EventStatus> findOne(Integer id) {
    return eventStatusRepository.findById(id);
  }
}

File: event-service/src/main/kotlin/com/greenlight/event/service/EventStatusService.kt
package com.greenlight.event.service

import com.greenlight.event.domain.EventStatus
import com.greenlight.event.error.NotFoundException
import com.greenlight.event.repository.EventStatusRepository
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.reactive.asFlow
import kotlinx.coroutines.reactive.awaitFirstOrElse
import org.springframework.stereotype.Service

@Service
class EventStatusService(private val eventStatusRepository: EventStatusRepository) : ReadOnlyService<EventStatus, Int> {

    override fun findAll(): Flow<EventStatus> {
        return eventStatusRepository.findAll().asFlow()
    }

    override suspend fun findOne(id: Int): EventStatus {
        return eventStatusRepository.findById(id).
            awaitFirstOrElse { throw NotFoundException("Status $id not found") }
    }
}
